llvm/lib/CodeGen/GlobalISel/MachineIRBuilder.cpp

```diff
@@ -772,8 +772,6 @@ MachineInstrBuilder MachineIRBuilder::buildShuffleVector(const DstOp &Res,
   LLT DstTy = Res.getLLTTy(*getMRI());
   LLT Src1Ty = Src1.getLLTTy(*getMRI());
   LLT Src2Ty = Src2.getLLTTy(*getMRI());
-  assert((size_t)(Src1Ty.getNumElements() + Src2Ty.getNumElements()) >=
```

Comment:
As outlined in the commit, there are a couple of locations in the AArch64 where this assertion doesn't hold. Therefore, I removed it to bring the build function in line with the real(?) world usage of ShuffleVector.

https://github.com/llvm/llvm-project/blob/5d45815473496db4b041a008e60be17bd78c06ae/llvm/test/CodeGen/AArch64/GlobalISel/combine-shufflevector.mir#L35
https://github.com/llvm/llvm-project/blob/5d45815473496db4b041a008e60be17bd78c06ae/llvm/test/CodeGen/AArch64/GlobalISel/legalize-insert-vector-elt.mir#L285
https://github.com/llvm/llvm-project/blob/5d45815473496db4b041a008e60be17bd78c06ae/llvm/test/CodeGen/AArch64/GlobalISel/prelegalizercombiner-shuffle-vector.mir#L192

Reply:
Why did this assert start failing with your PR? It seemed to work before.

Reply:
I/we build our LLVM for development with `-DLLVM_ENABLE_ASSERTIONS=On`. If an assert hits, my first thought is: Oh wait, I broke something. My thought is not: this is a legacy assert and I should remove it. Please don't remove asserts. You should think about fixing your code/contribution.

Reply:
You are completely right, that the code should be checked with asserts and that they are an important part of ensuring quality going further. Thanks for the reminder, I think that is never unwelcome. However, in this case my PR doesn't break the assertion per se. Rather the examples of `G_SHUFFLE_VECTOR`, I mentioned before don't follow the assertion. The code asserts that the sum of the vectors is larger than the mask, but for none of those are true. Since those patterns exist, I want to support them and since we have a `buildShuffleVector` I want to use it (instead of reimplementing it). 

Anyways, I am completely willing to drop the commit, you know much more than me about things like this. But my reason for removing it was to pull into line with how the opcode is used, not because I thought it was a legacy assert :)

Reply:
The source vectors each must be the same size, and the result number of elements must exactly match the number of elements in the mask. The assert doesn't quite match the machine verifier's rules for G_SHUFFLE_VECTOR